extension/src/plots/vega/util.ts

```diff
@@ -120,214 +115,11 @@ export type Encoding = {
-  spec: any
-): spec is { layer: { layer: [{ encoding: Record<string, unknown> }] }[] } =>
-  spec.layer?.[1]?.layer?.[0]?.encoding?.x &&
-  isEqual(spec.layer[1].layer[0].mark, {
-    color: 'gray',
-    type: 'rule'
-  })
-
-const patchShapeEncoding = (spec: TopLevelSpec, encoding: Encoding) => {
-  const update: EncodingUpdate = {
-    encoding
-  }
-
-  if (encoding.shape && specHasVerticalLineOnHover(spec)) {
-    update.layer = spec.layer
-    update.layer[1].layer[0].encoding.shape = null
-  }
-
-  return update
-}
-
-const getSpecEncodingUpdate = (
-  spec: TopLevelSpec,
-  {
-    color,
-    shape,
-    strokeDash
-  }: {
-    color?: ColorScale
-    shape?: ShapeEncoding
-    strokeDash?: StrokeDashEncoding
-  }
-): EncodingUpdate => {
-  const encoding: Encoding = {}
-  if (color) {
-    encoding.color = {
-      legend: { disable: true },
-      scale: color
-    }
-  }
-
-  if (strokeDash) {
-    encoding.strokeDash = {
-      ...strokeDash,
-      legend: {
-        disable: true
-      }
-    }
-  }
-
-  if (shape) {
-    encoding.shape = {
-      ...shape,
-      legend: {
-        disable: true
-      }
-    }
-    encoding.detail = { field: shape.field }
-  }
-
-  return patchShapeEncoding(spec, encoding)
-}
-
-const mergeUpdate = (spec: TopLevelSpec, update: EncodingUpdate) => {
-  let newSpec = cloneDeep(spec) as any
-
-  if (newSpec.concat?.length) {
-    newSpec.concat = newSpec.concat.map((c: any) => merge(c, update))
-  } else if (newSpec.hconcat?.length) {
-    newSpec.hconcat = newSpec.hconcat.map((c: any) => merge(c, update))
-  } else if (newSpec.vconcat?.length) {
-    newSpec.vconcat = newSpec.vconcat.map((c: any) => merge(c, update))
-  } else if (newSpec.spec) {
-    newSpec.spec = merge(newSpec.spec, update)
-  } else {
-    newSpec = merge(newSpec, update)
-  }
-
-  return newSpec
-}
-
-const truncateTitleString = (title: string, size: number) =>
-  truncate(title, size, 'left')
-
-const truncateTitleAsArrayOrString = (title: Text, size: number) => {
-  if (Array.isArray(title)) {
-    return title.map(line => truncateTitleString(line, size))
-  }
-  return truncateTitleString(title as unknown as string, size)
-}
-
-const truncateTitlePart = (
-  title: Title,
-  key: 'text' | 'subtitle',
-  size: number
-) => {
-  if (hasOwnProperty(title, key)) {
-    const text = title[key] as unknown as Text
-    title[key] = truncateTitleAsArrayOrString(text, size)
-  }
-}
-
-const truncateTitle = (
-  title: Title | Text | TitleParams<ExprRef | SignalRef> | undefined,
-  size: number
-) => {
-  if (!title) {
-    return ''
-  }
-
-  if (typeof title === 'string') {
-    return truncateTitleString(title, size)
-  }
-
-  if (Array.isArray(title)) {
-    return truncateTitleAsArrayOrString(title as Text, size)
-  }
-
-  const titleCopy = { ...title } as Title
-  truncateTitlePart(titleCopy, 'text', size)
-  truncateTitlePart(titleCopy, 'subtitle', size)
-  return titleCopy
-}
-
 export const truncateVerticalTitle = (
```

Comment:
[F] This is currently used for custom templates only. We know that the title is a string. I am going to migrate custom plots to use anchors as well so that we can use the same code for both in the frontend.